mindmeld/text_preparation/preprocessors.py

```diff
@@ -33,18 +33,37 @@ def process(self, text):
         """
         pass
 
-    @abstractmethod
-    def get_char_index_map(self, raw_text, processed_text):
-        """
-        Generates character index mapping from processed query to raw query.
 
-        See the Tokenizer class for a similar implementation.
+class NoOpPreprocessor(Preprocessor):
+    """
+    NoOpPreprocessor object
+    """
 
+    def process(self, text):
+        """
         Args:
-            raw_text (str)
-            processed_text (str)
+            text (str)
+
+        Returns:
+            (str)
+        """
+        return text
+
 
+class PreprocessorFactory:
+    """Preprocessor Factory Class"""
+
+    @staticmethod
+    def get_preprocessor(preprocessor: str):
+        """A static method to get a Preprocessor
+
+        Args:
+            preprocessor (str): Name of the desired Preprocessor class
         Returns:
-            (dict, dict): A tuple consisting of two maps, forward and backward
+            (Preprocessor): Preprocessor Class
         """
-        pass
+        preprocessor_classes = {NoOpPreprocessor.__name__: NoOpPreprocessor}
```

Comment:
I think these dicts can be added to a contants.py file

Reply:
For preprocessors.py, I changed this logic to a single if statement since there is only one preprocessor.

```
if preprocessor == NoOpPreprocessor.__name__:
    return NoOpPreprocessor()
raise TypeError(f"{preprocessor} is not a valid Preprocessor type.")
```

Would you like for us to add the class mapping dicts in other component files into constants.py?